apps/api/src/domain/models/resource.entity.ts

```diff
@@ -12,7 +14,15 @@ import BaseDomainModel from './BaseDomainModel';
 import { EdgeConnectionContext } from './context/context.entity';
 import { EdgeConnectionContextType } from './context/types/EdgeConnectionContextType';
 
+type Event = Record<string, unknown>;
+
 export abstract class Resource extends BaseDomainModel implements HasAggregateId {
+    /**
+     * We make this property optional so we don't need to specify it on existing data
+     * or test data. If it is not on a DTO, it will be set to [] in the constructor.
+     */
+    readonly events?: Event[];
```

Comment:
Woohoo!  Is there a way to make its optional status conditional on the circumstances?

Reply:
Also, just wondering about the word events.  I guess we should stick with this name in our backend and eventually come up with another name, or just a different nomenclature for the Events resource from frontend to backend